File: 11057.cpp
// 오르막 수
// https://www.acmicpc.net/problem/11057

#include <cstdio>
#include <vector>
using namespace std;

int main() {
	int n = 0;
	scanf("%d", &n);

	vector<vector<int> > dp = 
		vector<vector<int> >(10, vector<int>(n + 1, 0));

	for (int i = 0; i < 10; ++i) {
		dp[i][1] = 1;
		dp[i][0] = 0;
	}

	for (int i = 2; i <= n; ++i) {
		dp[0][i] = 1;
	}

	for (int i = 2; i <= n; ++i) {
		for (int j = 1; j < 10; ++j) {
			for (int k = 0; k <= j; ++k) {
				dp[j][i] = (dp[j][i] + dp[k][i - 1]) % 10007;
			}
		}
	}

	int answer = 0;
	for (int i = 0; i < 10; ++i) {
		answer = (answer + dp[i][n]) % 10007;
	}
	printf("%d\n", answer);

	return 0;
}

File: 11057.py
# 오르막 수
# https://www.acmicpc.net/problem/11057

import sys
from itertools import combinations_with_replacement

input = sys.stdin.readline

n = int(input())

dp = [[0] * n for _ in range(10)]

for i in range(10):
    dp[i][0] = 1

for i in range(1, n):
    for j in range(10):
        for k in range(j, 10):
            dp[j][i] += dp[k][i - 1]

answer = 0
for i in range(10):
    answer += dp[i][n - 1]
    answer %= 10007

print(answer)
